cmd/pilot-agent/main.go

```diff
@@ -29,9 +29,10 @@ import (
 )
 
 var (
-	configpath string
-	meshconfig string
-	role       proxy.Node
+	configpath   string
+	meshconfig   string
+	role         proxy.Node
+	customConfig string
```

Comment:
rather than adding another *config string, how about a simple boolean that says "generateConfig" ? You can reuse the configpath variable and populate it with the path to envoy config. Later in code, if generateConfig is true, generate config, else reuse config.